api/v1alpha1/nginxingresscontroller_types.go

```diff
@@ -72,9 +78,15 @@ type DefaultSSLCertificate struct {
 	// Secret in the form of a Key Vault URI
 	// +optional
 	KeyVaultURI *string `json:"keyVaultURI"`
+
+	// ForceSSLRedirect is a flag that sets the global value of redirects to HTTPS if there is a defined DefaultSSLCertificate
+	// +kubebuilder:default:=false
+	// forceSSLRedirect is set to false by default and will add the "forceSSLRedirect: false" property even if the user doesn't specify it.
+	// If a user adds both a keyvault uri and secret the property count will be 3 since forceSSLRedirect still automatically gets added thus failing the check.
+	ForceSSLRedirect bool `json:"forceSSLRedirect,omitempty"`
 }
 
-// Secret is a struct that holds a name and namespace to be used in DefaultSSLCertificate
+// Secret is a struct that holds a name and namespace with length checking on the crd
```

Comment:
revert this comment. CRD isn't very descriptive. These are essentially user docs for how to use the crd.